Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+RequestQueue.swift

```diff
@@ -0,0 +1,115 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+extension HTTPConnectionPool {
+    /// A struct to store all queued requests.
+    struct RequestQueue {
+        private var generalPurposeQueue: CircularBuffer<Request>
+        private var eventLoopQueues: [EventLoopID: CircularBuffer<Request>]
```

Comment:
Minor: this would be mildly nicer using queues of your own type rather than `CircularBuffer`

Reply:
Although it's really only the removal code that's duplicated. Don't mind whether you change it or not.